### PR TITLE
Adjust dashboard to be mobile friendly

### LINKED ISSUE #112: Adjust dashboard to be mobile friendly
Thoughts so far:

when screen width is smaller than tablet size (<768px):

- [ ] collapsed menu

- [ ] sidebar -> bottom drawer

- [ ] map popup -> submission in full view inside bottom drawer

### PR DESCRIPTION
**Pull request by hulloanson**
Key changes:

Main breakpoint is when screen size is smaller than a tablet (i.e. less than 1024px). At this smaller size:
- Sidebar is placed at the bottom
- Popups do not show as long as the screen height is 500px or greater
- Header links are collapsed into a hamburger menu

Also, in all views, the selected map pin is now displayed at a larger size than the other surrounding pins to help with differentiation

### CODE REVIEW

**Comment by tallcoleman** [comment]
This is already looking really cool @hulloanson !

**Comment by tallcoleman** [comment]
I'm not sure if we should remove the `&` symbol yet for nested CSS - e.g. for Chrome, the browser versions from between April and October 2023 still require it: https://caniuse.com/css-nesting

File: bikespace_dashboard/css/components/page/mobile.css [hunk 1]
```diff
@@ -0,0 +1,114 @@
+@import '../stylevars.css';
+
+:root {
+  /* preferred visible area of map.
+   * taller screens takes advantage of this when the panel is open
+   * to show a bigger map
```
Comment (tallcoleman) [inline bikespace_dashboard/css/components/page/mobile.css:6]: Does this make the map bigger or the panel? It looks like it increases the panel size when more space is available.

**Comment by tallcoleman** [comment]
# Notes from review:

## Must fix before merge

- [ ] Nav bar disappears on desktop view when a pin is focused
- [ ] Can't click on map to un-focus a pin
- [ ] Since the feed is the only data shown in mobile view, it should show all the relevant info for a report (need to add parking duration and issue id)

## Nice to have 

- [ ] Change drawer button to an up/down chevron to make interaction intent clearer
- [ ] Make map credits smaller in mobile view (e.g. [maplibre uses a toggle](https://maplibre.org/maplibre-gl-js/docs/examples/add-a-marker/))
- [ ] Sidebar content is very wide just below the mobile breakpoint - should consider either having a multi-column flex layout or making the breakpoint smaller

**Review by tallcoleman — COMMENTED** [review]
## Notes from review:

### Must fix before merge

- [x] Nav bar disappears on desktop view when a pin is focused
- [x] Can't click on map to un-focus a pin
- [x] Since the feed is the only data shown in mobile view, it should show all the relevant info for a report (need to add parking duration and issue id)

### Nice to have 

- [ ] Change drawer button to an up/down chevron to make interaction intent clearer
- [ ] Make map credits smaller in mobile view (e.g. [maplibre uses a toggle](https://maplibre.org/maplibre-gl-js/docs/examples/add-a-marker/))
- [ ] Sidebar content is very wide just below the mobile breakpoint - should consider either having a multi-column flex layout or making the breakpoint smaller

**Comment by tallcoleman** [comment]
@hulloanson "Can't click on map to un-focus a pin" from the review should now be resolved. Also made a number of edits to try to centralize all management of app state through the hash router.

**Comment by tallcoleman** [comment]
@hulloanson fixed this one, now just "Nav bar disappears on desktop view when a pin is focused" which needs fixing before merge

> Since the feed is the only data shown in mobile view, it should show all the relevant info for a report (need to add parking duration and issue id)

**Comment by hulloanson** [comment]
> @hulloanson fixed this one, now just "Nav bar disappears on desktop view when a pin is focused" which needs fixing before merge

fixed. check it out :)

**Review by tallcoleman — APPROVED** [review]
@hulloanson ready to merge I think! I just added some final tweaks. You were right about how challenging it is to get `scrollIntoView()` working properly - have added some changes to prevent it from hiding the feed title, but if you think there's a better way of fixing that problem, let me know.

**Pull request merged**